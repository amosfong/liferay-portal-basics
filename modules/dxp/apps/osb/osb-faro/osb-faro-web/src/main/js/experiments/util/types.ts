export type FormatYAxisFn = (metricUnit: string) => (value: number) => string;

export type GetFormattedMedianFn = (
	median: number,
	metric: MetricName
) => string;

export type GetMetricNameFn = (metric: MetricName) => string;

export type GetMetricUnitFn = (metric: MetricName) => string;

export type GetStatusColorFn = (status: Status) => string;

export type GetStatusNameFn = (status: Status) => string;

export type GetShortIntervals = (intervals: Array<Date>) => Array<Date>;

export type GetTicksFn = (maxValue: number) => Array<number>;

export type GetVariantLabels = (experiment: {
	bestVariant?: Variant;
	dxpVariantId: string;
	publishedDXPVariantId?: string;
	status: Status;
	winnerDXPVariantId?: string;
}) => Array<{status: string; value: string}>;

type MetricName =
	| 'BOUNCE_RATE'
	| 'CLICK_RATE'
	| 'MAX_SCROLL_DEPTH'
	| 'TIME_ON_PAGE';

type MergedVariant = {
	changes: number;
	confidenceInterval: Array<number>;
	control: boolean;
	dxpVariantId: string;
	dxpVariantName: string;
	improvement: number;
	median: number;
	probabilityToWin: number;
	trafficSplit: number;
	uniqueVisitors: number;
};

export type MergedVariantsFn = (
	variants: Array<Variant>,
	variantMetrics: Array<VariantMetric>
) => Array<MergedVariant>;

type Status =
	| 'COMPLETED'
	| 'DRAFT'
	| 'FINISHED_NO_WINNER'
	| 'FINISHED_WINNER'
	| 'PAUSED'
	| 'RUNNING'
	| 'SCHEDULED'
	| 'TERMINATED';

type Variant = {
	changes: number;
	control: boolean;
	dxpVariantId: string;
	dxpVariantName: string;
	trafficSplit: number;
	uniqueVisitors: number;
};

type VariantMetric = {
	confidenceInterval: Array<number>;
	dxpVariantId: string;
	improvement: number;
	median: number;
	probabilityToWin: number;
};
