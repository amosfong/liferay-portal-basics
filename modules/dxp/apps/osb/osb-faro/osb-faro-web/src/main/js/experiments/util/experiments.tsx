import {CHART_COLORS} from 'shared/util/charts';
import {CONTROL_COLOR} from './constants';
import {
	FormatYAxisFn,
	GetFormattedMedianFn,
	GetMetricNameFn,
	GetMetricUnitFn,
	GetShortIntervals,
	GetStatusColorFn,
	GetStatusNameFn,
	GetTicksFn,
	GetVariantLabels,
	MergedVariantsFn
} from './types';
import {round} from 'lodash';
import {toRounded, toThousands, toThousandsBase} from 'shared/util/numbers';

const METRICS_NAMES = new Map([
	['BOUNCE_RATE', Liferay.Language.get('bounce-rate')],
	['CLICK_RATE', Liferay.Language.get('click-through-rate')],
	['MAX_SCROLL_DEPTH', Liferay.Language.get('max-scroll-depth')],
	['TIME_ON_PAGE', Liferay.Language.get('view-duration')]
]);

const METRICS_UNITS = new Map([
	['BOUNCE_RATE', '%'],
	['CLICK_RATE', '%'],
	['MAX_SCROLL_DEPTH', '%'],
	['TIME_ON_PAGE', 's']
]);

const STATUS_COLORS = new Map([
	['COMPLETED', 'success'],
	['DRAFT', 'secondary'],
	['FINISHED_NO_WINNER', 'secondary'],
	['FINISHED_WINNER', 'success'],
	['PAUSED', 'secondary'],
	['RUNNING', 'info'],
	['SCHEDULED', 'warning'],
	['TERMINATED', 'danger']
]);

const STATUS_NAMES = new Map([
	['COMPLETED', Liferay.Language.get('complete')],
	['DRAFT', Liferay.Language.get('draft')],
	['FINISHED_NO_WINNER', Liferay.Language.get('no-winner')],
	['FINISHED_WINNER', Liferay.Language.get('winner-declared')],
	['PAUSED', Liferay.Language.get('paused')],
	['RUNNING', Liferay.Language.get('running')],
	['SCHEDULED', Liferay.Language.get('scheduled')],
	['TERMINATED', Liferay.Language.get('terminated')]
]);

const getExperimentLink = ({action, id, pageURL}) => {
	const experimentLink = `${pageURL}?segmentsExperimentKey=${id}`;

	if (action) {
		return `${experimentLink}&segmentsExperimentAction=${action}`;
	}

	return experimentLink;
};

export const formatYAxis: FormatYAxisFn = metricUnit => value => {
	if (value % 1 === 0) {
		return `${value}${metricUnit}`;
	}

	return `${value.toFixed(1)}${metricUnit}`;
};

export const getFormattedMedian: GetFormattedMedianFn = (median, metric) => {
	const precision = metric === 'CLICK_RATE' ? 3 : 2;

	return toRounded(median, precision);
};

export const getFormattedMedianLabel = metric =>
	metric === 'CLICK_RATE'
		? `${Liferay.Language.get('median')} ${getMetricName(metric)}`
		: `${getMetricName(metric)} ${Liferay.Language.get('median')}`;

export const getFormattedProbabilityToWin = value => {
	if (value < 0.1) {
		value = '< 0.1';
	} else if (value > 99.9) {
		value = '> 99.9';
	} else {
		value = toRounded(value);
	}

	return value;
};

export const getMetricName: GetMetricNameFn = metric =>
	METRICS_NAMES.get(metric);

export const getMetricUnit: GetMetricUnitFn = metric =>
	METRICS_UNITS.get(metric);

export const getStatusColor: GetStatusColorFn = status =>
	STATUS_COLORS.get(status);

export const getStatusName: GetStatusNameFn = status =>
	STATUS_NAMES.get(status).toUpperCase();

export const mergedVariants: MergedVariantsFn = (variants, variantMetrics) =>
	variants.map(variant => ({
		...variant,
		...variantMetrics.find(
			({dxpVariantId}) => variant.dxpVariantId === dxpVariantId
		)
	}));

export const getActions = (
	status: string,
	{id, onDelete, pageURL, publishable} = null
) => {
	const deleteButton = {
		displayType: 'secondary',
		label: Liferay.Language.get('delete'),
		onClick: onDelete
	};

	switch (status) {
		case 'COMPLETED': {
			return [deleteButton];
		}
		case 'DRAFT': {
			return [
				{
					displayType: 'primary',
					label: Liferay.Language.get('review'),
					redirectURL: getExperimentLink({
						action: 'reviewAndRun',
						id,
						pageURL
					})
				},
				{
					displayType: 'secondary',
					label: Liferay.Language.get('delete'),
					redirectURL: getExperimentLink({
						action: 'delete',
						id,
						pageURL
					})
				}
			];
		}
		case 'FINISHED_NO_WINNER':
		case 'FINISHED_WINNER': {
			return [
				{
					displayType: 'primary',
					label: Liferay.Language.get('publish'),
					redirectURL: getExperimentLink({
						action: 'publish',
						id,
						pageURL
					})
				},
				{
					displayType: 'secondary',
					label: Liferay.Language.get('delete'),
					redirectURL: getExperimentLink({
						action: 'delete',
						id,
						pageURL
					})
				}
			];
		}
		case 'TERMINATED': {
			if (publishable) {
				return [
					{
						displayType: 'primary',
						label: Liferay.Language.get('publish'),
						redirectURL: getExperimentLink({
							action: 'publish',
							id,
							pageURL
						})
					},
					{
						displayType: 'secondary',
						label: Liferay.Language.get('delete'),
						redirectURL: getExperimentLink({
							action: 'delete',
							id,
							pageURL
						})
					}
				];
			}

			return [deleteButton];
		}
		case 'RUNNING': {
			return [
				{
					displayType: 'secondary',
					label: Liferay.Language.get('terminate'),
					redirectURL: getExperimentLink({
						action: 'terminate',
						id,
						pageURL
					})
				}
			];
		}
		default: {
			return [];
		}
	}
};

export const getBestVariant = ({
	dxpVariants,
	goal,
	metrics: {variantMetrics}
}) => {
	if (
		!dxpVariants ||
		variantMetrics.every(({median}) => median === variantMetrics[0].median)
	) {
		return null;
	}

	if (goal?.metric === 'BOUNCE_RATE') {
		return mergedVariants(
			dxpVariants,
			variantMetrics
		).reduce((prev, current) =>
			prev.median < current.median ? prev : current
		);
	}

	return mergedVariants(dxpVariants, variantMetrics).reduce((prev, current) =>
		prev.median > current.median ? prev : current
	);
};

export const getVariantLabels: GetVariantLabels = ({
	bestVariant,
	dxpVariantId,
	publishedDXPVariantId,
	status,
	winnerDXPVariantId
}) => {
	const labels = [];

	if (status === 'RUNNING' && bestVariant?.dxpVariantId === dxpVariantId) {
		labels.push({
			status: 'success',
			value: Liferay.Language.get('current-best')
		});
	}

	if (
		winnerDXPVariantId === dxpVariantId &&
		(status === 'COMPLETED' || status === 'FINISHED_WINNER')
	) {
		labels.push({
			status: 'success',
			value: Liferay.Language.get('winner')
		});
	}

	if (publishedDXPVariantId === dxpVariantId) {
		labels.push({
			status: 'info',
			value: Liferay.Language.get('published')
		});
	}

	return labels;
};

export const getTicks: GetTicksFn = maxValue => {
	const arr = [];
	let interval = 1;
	const step = Math.round(maxValue / 8);

	while (interval <= maxValue) {
		arr.push(interval);

		interval = interval + step;
	}

	return [...arr];
};

export const getShortIntervals: GetShortIntervals = intervals =>
	getTicks(intervals.length).map(tick => intervals[tick - 1]);

export const toThousandsABTesting = number => {
	if (number > 1e4) {
		return toThousandsBase(number, factor =>
			Math.trunc(round(number * factor, 2))
		);
	}

	return toThousands(number);
};

export const getLegendData = dxpVariants => {
	const COLORS = [...CHART_COLORS];

	return dxpVariants.map(({control, dxpVariantId, dxpVariantName}) => ({
		color: control ? CONTROL_COLOR : COLORS.shift(),
		id: dxpVariantId,
		name: dxpVariantName
	}));
};

export const getMedianGraphData = ({dxpVariants, metricUnit}) => {
	const COLORS = [...CHART_COLORS];

	const type = metricUnit === '%' ? 'percentage' : 'number';

	const formatter =
		metricUnit === '%' ? value => value : value => `${value}s`;

	const items = dxpVariants.map(({confidenceInterval, control, median}) => ({
		intervals: [
			{
				end: confidenceInterval[1],
				start: confidenceInterval[0]
			}
		],
		progress: [
			{
				color: control ? CONTROL_COLOR : COLORS.shift(),
				value: median
			}
		]
	}));

	const maxValue = Math.max(
		...dxpVariants.map(({confidenceInterval}) => confidenceInterval[1])
	);

	return {
		disableScroll: true,
		empty: maxValue === 0,
		formatSpacement: false,
		grid: {
			formatter,
			maxValue,
			minValue: 0,
			precision: 2,
			show: true,
			type
		},
		items
	};
};
