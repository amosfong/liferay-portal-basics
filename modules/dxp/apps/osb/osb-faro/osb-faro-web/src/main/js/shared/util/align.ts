export const POSITIONS = {
	BottomCenter: 4,
	BottomLeft: 5,
	BottomRight: 3,
	LeftCenter: 6,
	RightCenter: 2,
	TopCenter: 0,
	TopLeft: 7,
	TopRight: 1
};

export type Position = typeof POSITIONS[keyof typeof POSITIONS];

export function align(element, alignElement, position, autoBestAlign = true) {
	let bestRegion;

	if (autoBestAlign) {
		const suggestion = suggestAlignBestRegion(
			element,
			alignElement,
			position
		);
		position = suggestion.position;
		bestRegion = suggestion.region;
	} else {
		bestRegion = getAlignRegion(element, alignElement, position);
	}

	const computedStyle = window.getComputedStyle(element, null);

	if (computedStyle.getPropertyValue('position') !== 'fixed') {
		bestRegion.top += window.scrollY;
		bestRegion.left += window.scrollX;

		let offsetParent = element;
		while ((offsetParent = offsetParent.offsetParent)) {
			bestRegion.top -= offsetParent.offsetTop;
			bestRegion.left -= offsetParent.offsetLeft;
		}
	}

	element.style.top = `${bestRegion.top}px`;
	element.style.left = `${bestRegion.left}px`;

	return position;
}

function getAlignRegion(element, alignElement, position) {
	const r1 = alignElement.getBoundingClientRect();
	const r2 = element.getBoundingClientRect();
	let top = 0;
	let left = 0;

	switch (position) {
		case POSITIONS.TopCenter:
		default:
			top = r1.top - r2.height;
			left = r1.left + r1.width / 2 - r2.width / 2;
			break;
		case POSITIONS.RightCenter:
			top = r1.top + r1.height / 2 - r2.height / 2;
			left = r1.left + r1.width;
			break;
		case POSITIONS.BottomCenter:
			top = r1.bottom;
			left = r1.left + r1.width / 2 - r2.width / 2;
			break;
		case POSITIONS.LeftCenter:
			top = r1.top + r1.height / 2 - r2.height / 2;
			left = r1.left - r2.width;
			break;
		case POSITIONS.TopRight:
			top = r1.top - r2.height;
			left = r1.right - r2.width;
			break;
		case POSITIONS.BottomRight:
			top = r1.bottom;
			left = r1.right - r2.width;
			break;
		case POSITIONS.BottomLeft:
			top = r1.bottom;
			left = r1.left;
			break;
		case POSITIONS.TopLeft:
			top = r1.top - r2.height;
			left = r1.left;
			break;
	}

	return {
		bottom: top + r2.height,
		height: r2.height,
		left,
		right: left + r2.width,
		top,
		width: r2.width
	};
}

function suggestAlignBestRegion(element, alignElement, position) {
	let bestArea = 0;
	let bestPosition = position;
	let bestRegion = getAlignRegion(element, alignElement, bestPosition);
	let tryPosition = bestPosition;
	let tryRegion = bestRegion;

	const viewportRegion = getViewportRegion();

	for (let i = 0; i < 8; ) {
		if (intersectRegion(viewportRegion, tryRegion)) {
			const visibleRegion = intersection(viewportRegion, tryRegion);

			const area = visibleRegion.width * visibleRegion.height;

			if (area > bestArea) {
				bestArea = area;
				bestRegion = tryRegion;
				bestPosition = tryPosition;
			}

			if (insideRegion(viewportRegion, tryRegion)) {
				break;
			}
		}
		tryPosition = (position + ++i) % 8;
		tryRegion = getAlignRegion(element, alignElement, tryPosition);
	}

	return {
		position: bestPosition,
		region: bestRegion
	};
}

function getViewportRegion() {
	const height = document.documentElement.offsetHeight;
	const width = document.documentElement.offsetWidth;

	return makeRegion(height, height, 0, width, 0, width);
}

function insideRegion(r1, r2) {
	return (
		r2.top >= r1.top &&
		r2.bottom <= r1.bottom &&
		r2.right <= r1.right &&
		r2.left >= r1.left
	);
}

function intersectRect(x0, y0, x1, y1, x2, y2, x3, y3) {
	return !(x2 > x1 || x3 < x0 || y2 > y1 || y3 < y0);
}

function intersectRegion(r1, r2) {
	return intersectRect(
		r1.top,
		r1.left,
		r1.bottom,
		r1.right,
		r2.top,
		r2.left,
		r2.bottom,
		r2.right
	);
}

function intersection(r1, r2) {
	if (!intersectRegion(r1, r2)) {
		return null;
	}
	const bottom = Math.min(r1.bottom, r2.bottom);
	const right = Math.min(r1.right, r2.right);
	const left = Math.max(r1.left, r2.left);
	const top = Math.max(r1.top, r2.top);

	return makeRegion(bottom, bottom - top, left, right, top, right - left);
}

function makeRegion(bottom, height, left, right, top, width) {
	return {
		bottom,
		height,
		left,
		right,
		top,
		width
	};
}
