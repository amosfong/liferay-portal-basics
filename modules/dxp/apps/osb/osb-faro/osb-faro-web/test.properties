##
## Test Suites
##

    modified.files.includes[relevant][osb-faro-web-playwright-rule]=**
    relevant.rule.names=osb-faro-web-playwright-rule

    test.batch.names[relevant][osb-faro-web-playwright-rule]=\
        playwright-js-tomcat90-mysql57

    #
    # Relevant
    #

    playwright.projects.includes[playwright-js-tomcat90-mysql57][relevant][osb-faro-web-playwright-rule]=\
        osb-faro-web

##
## Testray
##

    testray.main.component.name=Analytics Cloud