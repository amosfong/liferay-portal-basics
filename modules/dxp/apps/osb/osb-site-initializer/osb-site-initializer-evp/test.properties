##
## Test Suites
##

    modified.files.includes[relevant][osb-site-initializer-evp-functional-rule]=**
    relevant.rule.names=osb-site-initializer-evp-functional-rule

    test.batch.names[relevant][osb-site-initializer-evp-functional-rule]=\
        functional-tomcat90-mysql57

    #
    # Portal Hotfix Release
    #

    test.batch.run.property.query[functional-bundle-tomcat-mysql57][test-portal-hotfix-release]=\
        ${test.batch.run.property.query[functional-tomcat*-mysql*][relevant][osb-site-initializer-evp-functional-rule]}

    #
    # Relevant
    #

    test.batch.run.property.query[functional-tomcat*-mysql*][relevant][osb-site-initializer-evp-functional-rule]=\
        (portal.acceptance == true) AND \
        (\
            (testray.component.names ~ "OSB Site Initializer EVP") OR \
            (testray.main.component.name ~ "OSB Site Initializer EVP")\
        )

##
## Testray
##

    testray.main.component.name=OSB Site Initializer EVP