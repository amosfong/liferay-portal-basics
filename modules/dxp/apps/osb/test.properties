##
## Test Suites
##

    modified.files.includes[relevant][osb-java-rule]=\
        **/*.java,\
        **/test/**,\
        **/testIntegration/**

    relevant.rule.names=osb-java-rule

    test.batch.names[relevant][osb-java-rule]=\
        modules-integration-mysql57,\
        modules-unit

    #
    # Relevant
    #

    modules.includes.required.test.batch.class.names.includes[modules-integration-mysql57][relevant][osb-java-rule]=\
        dxp/apps/osb/**/*Test.java

    modules.includes.required.test.batch.class.names.includes[modules-unit][relevant][osb-java-rule]=\
        dxp/apps/osb/**/*Test.java

##
## Testray
##

    testray.main.component.name=Analytics Cloud