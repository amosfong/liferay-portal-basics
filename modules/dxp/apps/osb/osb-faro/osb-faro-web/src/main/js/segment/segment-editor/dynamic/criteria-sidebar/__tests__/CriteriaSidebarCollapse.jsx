import * as data from 'test/data';
import CriteriaSidebarCollapse from '../CriteriaSidebarCollapse';
import React from 'react';
import {cleanup, render} from '@testing-library/react';
import {DndProvider} from 'react-dnd';
import {HTML5Backend} from 'react-dnd-html5-backend';
import {List} from 'immutable';
import {Property, PropertyGroup, PropertySubgroup} from 'shared/util/records';

jest.unmock('react-dom');

describe('CriteriaSidebarCollapse', () => {
	const propertyGroupsIList = new List([
		new PropertyGroup({
			label: 'Interests',
			name: 'Interests',
			propertyKey: 'interests',
			propertySubgroups: new List([
				new PropertySubgroup({
					properties: new List([
						data.getImmutableMock(Property, data.mockProperty, 0, {
							label: 'Page Views',
							name: 'Page Views'
						})
					])
				}),
				new PropertySubgroup({
					label: 'DXP Custom Fields',
					properties: new List([
						data.getImmutableMock(Property, data.mockProperty, 0, {
							label: 'Page Actions',
							name: 'Page Actions'
						})
					])
				})
			])
		})
	]);

	const propertyGroupsIListCustomEvents = new List([
		new PropertyGroup({
			label: 'Events',
			name: 'Events',
			propertyKey: 'events',
			propertySubgroups: new List([
				new PropertySubgroup({
					label: 'Custom Events',
					properties: new List([
						data.getImmutableMock(Property, data.mockProperty, 0, {
							label: 'Custom Event 1',
							name: 'Custom Event 1'
						}),

						data.getImmutableMock(Property, data.mockProperty, 0, {
							label: 'Custom Event 2',
							name: 'Custom Event 2'
						}),

						data.getImmutableMock(Property, data.mockProperty, 0, {
							label: 'Custom Event 3',
							name: 'Custom Event 3'
						})
					])
				})
			])
		})
	]);

	afterEach(cleanup);

	it('should render', () => {
		const {container} = render(
			<DndProvider backend={HTML5Backend}>
				<CriteriaSidebarCollapse
					propertyGroupsIList={propertyGroupsIList}
					propertyKey='interests'
				/>
			</DndProvider>
		);

		expect(container).toMatchSnapshot();
	});

	it('should render w/ no results for only one subgroup', () => {
		const {queryByText} = render(
			<DndProvider backend={HTML5Backend}>
				<CriteriaSidebarCollapse
					propertyGroupsIList={propertyGroupsIList}
					propertyKey='interests'
					searchValue='Actions'
				/>
			</DndProvider>
		);

		expect(queryByText('No results were found.')).toBeTruthy();
		expect(queryByText('Page Views')).toBeNull();
		expect(queryByText('Page Actions')).toBeTruthy();
	});

	it('should render w/ no results', () => {
		const {queryByText} = render(
			<DndProvider backend={HTML5Backend}>
				<CriteriaSidebarCollapse
					propertyGroupsIList={propertyGroupsIList}
					searchValue='should not exist'
				/>
			</DndProvider>
		);

		expect(queryByText('No results were found.')).toBeTruthy();
		expect(queryByText('Page Views')).toBeNull();
		expect(queryByText('Page Actions')).toBeNull();
	});

	it('should render custom events in the sidebar', () => {
		const {queryByText} = render(
			<DndProvider backend={HTML5Backend}>
				<CriteriaSidebarCollapse
					propertyGroupsIList={propertyGroupsIListCustomEvents}
					propertyKey='events'
				/>
			</DndProvider>
		);

		expect(queryByText('Custom Events')).toBeInTheDocument();
		expect(queryByText('Custom Event 1')).toBeInTheDocument();
		expect(queryByText('Custom Event 2')).toBeInTheDocument();
		expect(queryByText('Custom Event 3')).toBeInTheDocument();
	});
});
