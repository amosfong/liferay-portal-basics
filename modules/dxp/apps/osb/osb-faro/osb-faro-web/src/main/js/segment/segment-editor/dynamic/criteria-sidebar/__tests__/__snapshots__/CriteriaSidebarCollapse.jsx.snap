// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CriteriaSidebarCollapse should render 1`] = `
<div>
  <ul
    class="property-subgroups-list active"
  >
    <li>
      
      <ul
        class="properties-list"
      >
        <li
          class="criteria-sidebar-item-root color--"
          data-testid="criteria-item-Page Views"
          draggable="true"
        >
          <span
            class="inline-item"
          >
            <svg
              class="lexicon-icon lexicon-icon-drag icon-root"
              role="presentation"
            >
              <use
                href="#drag"
              />
            </svg>
          </span>
          <span
            class="criteria-sidebar-item-type sticker"
          >
            <span
              class="inline-item"
            >
              <svg
                class="lexicon-icon lexicon-icon-text icon-root"
                role="presentation"
              >
                <use
                  href="#text"
                />
              </svg>
            </span>
          </span>
          Page Views
        </li>
      </ul>
    </li>
    <li>
      <div
        class="property-subgroup-label"
      >
        DXP Custom Fields
      </div>
      <ul
        class="properties-list"
      >
        <li
          class="criteria-sidebar-item-root color--"
          data-testid="criteria-item-Page Actions"
          draggable="true"
        >
          <span
            class="inline-item"
          >
            <svg
              class="lexicon-icon lexicon-icon-drag icon-root"
              role="presentation"
            >
              <use
                href="#drag"
              />
            </svg>
          </span>
          <span
            class="criteria-sidebar-item-type sticker"
          >
            <span
              class="inline-item"
            >
              <svg
                class="lexicon-icon lexicon-icon-text icon-root"
                role="presentation"
              >
                <use
                  href="#text"
                />
              </svg>
            </span>
          </span>
          Page Actions
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
